src/main/java/org/springframework/data/util/IterableUtils.java

```diff
@@ -0,0 +1,70 @@
+/*
+ * Copyright 2012-2018 the original author or authors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.springframework.data.util;
+
+import java.util.Collections;
+
+import org.springframework.lang.NonNull;
+import org.springframework.lang.Nullable;
+
+/**
+ * Utility class for working with {@link Iterable} objects.
+ *
+ * @author John Blum
+ * @see java.lang.Iterable
+ * @since 2.1.0
+ */
+public abstract class IterableUtils {
```

Comment:
Not sure that Spring Data is the best place for such a utility. I'd suggest to move it to Spring Framework.